test/Dialect/structured_fuse.mlir

```diff
@@ -0,0 +1,127 @@
+// CHECK:   %[[VAL1:.+]] = scf.for %arg1 = %[[C0]] to %[[C32]] step %[[C1]] iter_args(%arg2 = %[[VAL0]]) -> (tensor<32x16xf32>) {
+// CHECK:     %[[VAL2:.+]] = scf.for %arg3 = %[[C0]] to %[[C16]] step %[[C1]] iter_args(%arg4 = %arg2) -> (tensor<32x16xf32>) {
+// CHECK:       %[[VAL3:.+]] = arith.addi %arg1, %[[C3]] : index
+// CHECK:       %[[MUL1:.+]] = arith.muli %arg3, %[[C2]] : index
+// CHECK:       %[[VAL4:.+]] = arith.addi %[[MUL1]], %[[C5]] : index
+// CHECK:       %[[SLICE0:.+]] = tensor.extract_slice %[[ARG0]][%[[VAL3]], %[[VAL4]]] [1, 2] [1, 1] : tensor<40x40xf32> to tensor<1x2xf32>
+// CHECK:       %[[VAL5:.+]] = tensor.empty() : tensor<1x2xf32>
+// CHECK:       %[[VAL6:.+]] = linalg.elemwise_binary ins(%[[SLICE0]], %[[SLICE0]] : tensor<1x2xf32>, tensor<1x2xf32>) outs(%[[VAL5]] : tensor<1x2xf32>) -> tensor<1x2xf32>
+// CHECK:       %[[SLICE1:.+]] = tcp.slice %[[VAL6]] starts(%[[C0]], %[[C0]]) sizes(%[[C1]], %[[C1]]) strides(%[[C1]], %[[C2]]) : tensor<1x2xf32> -> tensor<1x1xf32>
+// CHECK:       %[[SLICE2:.+]] = tensor.extract_slice %arg4[%arg1, %arg3] [1, 1] [1, 1] : tensor<32x16xf32> to tensor<1x1xf32>
+// CHECK:       %[[UNARY:.+]] = linalg.elemwise_unary ins(%[[SLICE1]] : tensor<1x1xf32>) outs(%[[SLICE2]] : tensor<1x1xf32>) -> tensor<1x1xf32>
+// CHECK:       %[[ISLICE:.+]] = tensor.insert_slice %[[UNARY]] into %arg4[%arg1, %arg3] [1, 1] [1, 1] : tensor<1x1xf32> into tensor<32x16xf32>
+// CHECK:       scf.yield %[[ISLICE]] : tensor<32x16xf32>
+// CHECK:     }
+// CHECK:     scf.yield %[[VAL2]] : tensor<32x16xf32>
+// CHECK:   }
+// CHECK:   return %[[VAL1]] : tensor<32x16xf32>
+func.func @fuse_tcp_slice_stride2(%arg0: tensor<40x40xf32>) -> tensor<32x16xf32> {
+    %shape40 = tensor.empty() : tensor<40x40xf32>
+
+    %0 = linalg.elemwise_binary ins(%arg0, %arg0 : tensor<40x40xf32>, tensor<40x40xf32>)
+                             outs(%shape40: tensor<40x40xf32>) -> tensor<40x40xf32>
+
+    %c32 = arith.constant 32 : index
+    %c16 = arith.constant 16 : index
+    %c3 = arith.constant 3 : index
+    %c5 = arith.constant 5 : index
+    %c1 = arith.constant 1 : index
+    %c2 = arith.constant 2 : index
+    %slice = tcp.slice %0 starts ( %c3, %c5 ) sizes ( %c32, %c16 ) strides ( %c1, %c2 ) : tensor<40x40xf32> -> tensor<32x16xf32>
+
+    %shape = tensor.empty() : tensor<32x16xf32>
+    %ret = linalg.elemwise_unary ins(%slice: tensor<32x16xf32>) outs(%shape: tensor<32x16xf32>) -> tensor<32x16xf32>
+
+    return %ret : tensor<32x16xf32>
+}
+
+module attributes {transform.with_named_sequence} {
+  transform.named_sequence @__transform_main(%arg0: !transform.any_op {transform.readonly}) {
+    %slice = transform.structured.match ops{["tcp.slice"]} in %arg0 : (!transform.any_op) -> !transform.any_op
+    %unary = transform.structured.match ops{["linalg.elemwise_unary"]} in %arg0 : (!transform.any_op) -> !transform.any_op
+
+    %1, %loops:2 = transform.structured.fuse %unary {tile_sizes = [1, 1], tile_interchange = [0, 1]}
```

Comment:
I think it might be good to either modify this test or add one more test where the `tile_sizes` are not `[1, 1]`. That will ensure that we stress the case of the non-unit stride a bit more. Might also be good to have a case where we have back to back `tcp.slice` ops with non-trivial strides and a non-trivial tile size just to be additionally safe.

Reply:
Added tests for these 2 cases

Reply:
Thank you!